### PR TITLE
Packaging: OS X Installer Can Upgrade CVMFS

### PR DESCRIPTION
**Pull request by reneme**
This should make the new OS X installer clever enough to do a clean transition between CernVM-FS 2.1.x and CernVM-FS 2.2.0. Please review carefully. :-)

Testing should also be done, but will be somewhat manual I fear.

### CODE REVIEW

**Comment by reneme** [comment]
moved the `/usr`->`/usr/local` migration code to `cvmfs_server setup`. It's only run on Mac OS of course.

**Comment by reneme** [comment]
ready for review... When built on OS X 10.10, the OpenSSL libraries on 10.11 are properly picked up.

File: cvmfs/cvmfs_config [hunk 1]
```diff
@@ -1624,6 +1627,131 @@ runas() {
...
+      echo "found and quarantined legacy configuration file '$conf_file'"
+    fi
+  done
+
+  # rescue all configuration files that have been added by the user
+  local cwd="$(pwd)"
+  cd "$old_conf_dir" # remove $old_conf_dir prefix from `find` output
+  for conf_file in $(find . -type f); do
+    echo "found and rescued user added configuration file '$conf_file'"
+    mv -f $conf_file ${new_conf_dir}/${conf_file}
```
Comment (jblomer) [inline cvmfs/cvmfs_config]: Here it might be necessary to create the base directory in `${new_conf_dir}` first

Comment (reneme) [inline cvmfs/cvmfs_config]: good catch! Thanks.

**Comment by reneme** [comment]
Introduced the recommended changes both in the installer's conclusion-screen and the setup script.

**Pull request merged**